### ISSUE TITLE
spotify play --playlist

### ISSUE THREAD

**Issue by brianstrauch** [brianstrauch/spotify-cli#27]

**Comment by Threpio** [comment]
Is anyone working on the playing of a playlist? 

Also curious as to see if there is the possibility of implementing the ability to see the 'Liked songs' playlist (I use that for my spotify usage.

**Comment by brianstrauch** [comment]
Created an issue for playing the "Liked Songs" playlist! 😄 Although, this issue has to be completed first. Let me know if you're interested in working on it and I'll assign it to you.

**Comment by Threpio** [comment]
Was having a look at the codebase whilst I was curious - Not 100% sure I know how to apply it straight up. The generic Spotify libraries don't seem to support referencing a playlist as a command/variable. 
If I think I can take it on I will ping and take it - For the meanwhile it might be worth leaving it to someone a bit more experienced.

**Comment by Threpio** [comment]
I didn't realise that you were referencing your own Spotify go api library. Do we need to create the 'play playlist' functionality there and then update the reference/commands here? Sorry for the time

**Comment by brianstrauch** [comment]
No worries, the only change you'd have to implement in the library is searching by playlist/album (which should be relatively simple; just a few lines of code). It currently hardcodes "track" in the search parameters: https://github.com/brianstrauch/spotify/blob/5e406af4be1b0cc7d6fb0cdec0288bb09013d77c/search.go#L13

**Comment by Threpio** [comment]
<img width="1261" alt="Screenshot 2021-07-16 at 17 44 11" src="https://user-images.githubusercontent.com/49570057/125974031-b0c97376-53ec-44a6-9376-a7182ace8113.png">
A first stab at the changes to the spotify library - This could then be called in here. I think I was just going to reference the 'GetPlaylist' Function that had already been created?

**Comment by brianstrauch** [comment]
Actually... I think we can just use the existing `api.Play()` function! That function takes a variable number of `uris`, so you could probably just pass the URI of a playlist into the function and it would play that playlist. We would just have to make a new `api.SearchForPlaylist()` function to return a URI for a given playlist name.

**Comment by Threpio** [comment]
the GetPlaylist function returns a Playlist object - Which I believe contains the URI? 

`playlist.Meta.URI`

**Comment by brianstrauch** [comment]
@Threpio Yep, exactly!

**Comment by brianstrauch** [comment]
@Threpio Hmm... I was originally thinking we could give people the option to play *any* playlist, but being able to play only your own playlists is probably better for UX. In that case, we should call the `api.GetPlaylists()` function and do a case-insensitive search for the playlist name the user gave us. If we find it, we can grab the `URI` value and pass that to the `Play()` function.

So, the good news is that we won't have to implement anything new in the SDK, just the CLI. 😄

**Comment by brianstrauch** [comment]
Although the `--album` flag *will* need to implement a new `Search()` function. I'll split that off into a new issue

**Comment by Threpio** [comment]
--Album can be for a future PR.

I won't be able to do this this evening but am happy to look at it this weekend :)

**Comment by Threpio** [comment]
<img width="816" alt="Screenshot 2021-07-17 at 10 46 22" src="https://user-images.githubusercontent.com/49570057/126031642-bb0f0449-f2d2-4655-adbe-eddbc3025483.png">
I believe that a small change will be needed to the spotify library and it can't just be done here. The api.Play() function can take uri's of songs but a playlist or album requires a 'context_uri' field that we do not currently send?

**Comment by brianstrauch** [comment]
@Threpio Commented on your PR, but we can add `context_uri` to the existing `Play()` function.
